src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/HMPPSAuthClient.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.client
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import org.springframework.http.RequestEntity
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.exchange
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.config.AuthAwareAuthenticationToken
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.hmppsauth.GetTokenResponse
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.InternalServerErrorProblem
+import java.net.URI
+
+@Component
+class HMPPSAuthClient(
```

Comment:
I'm trying to picture how this will work if you have multiple client credentials? e.g one for each downstream service

Reply:
Is this a done thing in the HMPPS Auth world?  Guessing a least privilege thing?

Reply:
Yes, since different APIs require different roles and scopes. And roles are not (technically) unique across services. So it is best practice - HOWEVER IT IS AN UNWRITTEN BEST PRACTICE